core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java

```diff
@@ -1662,4 +1662,34 @@ public String toString() {
                 clientAddressSources, clientAddressTrustedProxyFilter, clientAddressFilter, clientAddressMapper,
                 enableServerHeader, enableDateHeader);
     }
+
+    ServerConfig buildServerConfig(ServerConfig existingConfig) {
+        final AnnotatedServiceExtensions extensions =
+                virtualHostTemplate.annotatedServiceExtensions();
+
+        assert extensions != null;
+
+        final VirtualHost defaultVirtualHost =
+                defaultVirtualHostBuilder.build(virtualHostTemplate);
+        final List<VirtualHost> virtualHosts =
+                virtualHostBuilders.stream()
+                        .map(vhb -> vhb.build(virtualHostTemplate))
+                        .collect(toImmutableList());
+        // Pre-populate the domain name mapping for later matching.
+        final Mapping<String, SslContext> sslContexts;
+        final SslContext defaultSslContext = findDefaultSslContext(defaultVirtualHost, virtualHosts);
+
+        return new ServerConfig(
```

Comment:
We need to check if `pingIntervalMillis`, `maxConnectionAgeMillis` and `idleTimeoutMillis` need to be adjusted.
Could you extract the same logic from https://github.com/line/armeria/blob/master/core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java#L1502-L1576
and apply that for this method as well? 😄

Reply:
Done